src/stat.h

```diff
@@ -483,10 +483,10 @@ template<class NUM> class MinValMax
   // happens when results are moved from a faster series to a slower one.
   MinValMax& operator+=(const MinValMax& rhs)
     {
-      //if (rhs.max > max) max=rhs.max;
-      //if (rhs.min < min) min=rhs.min;
-      max += rhs.max;
-      min += rhs.min;
+      if (rhs.max > max) max=rhs.max;
+      if (rhs.min < min) min=rhs.min;
+//      max += rhs.max;
+//      min += rhs.min;
       val += rhs.val;
```

Comment:
WRT your changes, good catch.  But also, when rhs.val is added to val, the min and max may need updating.  So I think we also need the following after line 490:
if (max < val) max=val;
if (min > val) min=val;